GANDLF/losses.py

```diff
@@ -61,7 +65,12 @@ def MCD(pm, gt, num_class, weights = None, ignore_class = None):
         
         if calculate_dice_for_label:
             currentDice = dice(gt[:, i, ...], pm[:, i, ...])
-            # currentDiceLoss = 1 - currentDice # subtract from 1 because this is a loss
+            if loss_type == 1:
+                currentDice = 1 - currentDice # subtract from 1 because this is a loss
+            elif loss_type == 2:
+                currentDice = -torch.log(
+                    currentDice + torch.finfo(torch.float32).eps
+                )  # negative because we want positive losses
```

Comment:
Not a fan of this. maybe we can provide log_dice? This makes things unclear.

Reply:
I just didn't want to have 2 implementations of dice loss floating around. This is the base function, in any case, so I figured it would be okay. What do you think?